### PR TITLE
[df] Add ChangeBeginAndEndEntries

### PR DESCRIPTION
**Pull request by kristupaspranc**

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 1]
```diff
@@ -87,6 +87,7 @@ namespace Internal {
 namespace RDF {
 class GraphCreatorHelper;
 void ChangeEmptyEntryRange(const ROOT::RDF::RNode &node, std::pair<ULong64_t, ULong64_t> &&newRange);
+void ChangeEntryRange(const RNode &node, const ULong64_t start, const ULong64_t end);
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:90]: I would name this function (and the related one in RLoopManager) `ChangeBeginAndEndEntries` to be more specific of the different use case involved here. At some point we will need to decide how to provide a more coherent API for all the similar use cases

**Review by vepadulano — CHANGES_REQUESTED** [review]

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 2]
```diff
@@ -87,6 +87,7 @@ namespace Internal {
 namespace RDF {
 class GraphCreatorHelper;
 void ChangeEmptyEntryRange(const ROOT::RDF::RNode &node, std::pair<ULong64_t, ULong64_t> &&newRange);
+void ChangeBeginAndEndEntries(const RNode &node, ULong64_t start, ULong64_t end);
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:90]: The data members are `fBeginEntry` and `fEndEntry` so let's change the name of the function parameter for symmetry
```suggestion
void ChangeBeginAndEndEntries(const RNode &node, ULong64_t begin, ULong64_t end);
```

**Comment by kristupaspranc** [comment]
Made the requested changes, added a test.
This function has nothing to do with `Foreach` as previously reported.
It only works with dataframes that are made out of `TTree` or `TChain` as intended, thus, it does not with a dataframe that is `Cache`d.

**Comment by vepadulano** [comment]
Changes are implemented in master

**Pull request closed without merging**